src/services/catalog.service.ts

```diff
@@ -145,26 +145,52 @@ export class ImageItem implements IServiceItem {
   public description: string;
   public longDescription: string;
   public resource: any;
-  public isBuilder: boolean;
-  public isHidden: boolean;
+  public builderSpecTagName: any;
   private tags: any;
   private catalogSrv: CatalogService;
 
   constructor (image: any, catalogSrv : CatalogService) {
     this.resource = image;
     this.catalogSrv = catalogSrv;
-    this.tags = this.catalogSrv.$filter('imageStreamTagTags')(this.resource);
-    this.iconClass = this.getIcon();
-    this.name = this.getName();
-    this.description = this.getDescription();
-    this.longDescription = this.getLongDescription();
-    this.catsBySubCats = this.getCategoriesBySubCategories();
-    this.isBuilder = _.includes(this.tags, 'builder');
-    this.isHidden = _.includes(this.tags, 'hidden');
+    this.builderSpecTagName = this.getbuilderSpecTagName();
+    if (this.builderSpecTagName) {
+      this.tags = this.getTags();
+      this.iconClass = this.getIcon();
+      this.name = this.getName();
+      this.description = this.getDescription();
+      this.longDescription = this.getLongDescription();
+      this.catsBySubCats = this.getCategoriesBySubCategories();
+    }
+  }
+
+  // A valid specTag has a 'builder' tag. no 'hidden' tag, and exists in 'status.tags'.
+  private getbuilderSpecTagName() {
+    let validSpecTag: any;
+
+    if (!this.resource.status) {
+      return null;
+    }
+
+    if (this.resource.spec && this.resource.spec.tags) {
+      validSpecTag = _.find(this.resource.spec.tags, (specTag: any) => {
+        let specTagTags: any = _.get(specTag, 'annotations.tags');
+        if (specTagTags && _.includes(specTagTags, 'builder') && !_.includes(specTagTags, 'hidden')) {
```

Comment:
I just noticed you're doing a substring match, which isn't technically right. It would also match tag `notabuilder`.

Really we should be splitting the tags string, e.g. `specTagTags.split(/\s*,\s*/)`, and checking if builder and hidden are in the array that's returned.